### PR TITLE
Replace custom gsconfig with geoserver-restconfig

### PR DESCRIPTION
**Pull request by swainn**
Replaces the custom gsconfig dependency with a community maintained package geoserver-restconfig for a more maintainable approach to Python 3 support in TDS.

Also included in this PR:

- Move `tests` dir out of `tethys_dataset_services` package
- Automated testing using tox
- Run tests with pytest (though still mostly implemented using unittest)
- Replaces Travis and Appveyor CI with GitHub Actions
- Updated README with instructions for running tests
- Migrated to setuptools new setup.cfg approach for packaging and install
- Re-implemented create_styles with ATCore version, b/c gsconfig API doesn't work.
- Refactored get_layer_group to use gsconfig get_layergroups instead of get_layergroup b/c of a bug with the latter that returns only None when workspace is None.

Note: next phase is to incorporate more of GSAPI from ATCore.

### CODE REVIEW

**Comment by swainn** [comment]
@sdc50 this is ready for review. It's not as daunting as it looks. Most of the changes are file moves -- I moved the tests dir out of the main package to the root level.

File: .github/workflows/e2e_tests.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+jobs:
+  e2e-test:
+    name: E2E Test Python ${{ matrix.py }} - Postgis ${{ matrix.postgis }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        py:
+          - "3.10"
+          # - "3.9"
```
Comment (gagelarsen) [inline .github/workflows/e2e_tests.yml:19]: Are we keeping these around for a reason?

Comment (swainn) [reply .github/workflows/e2e_tests.yml:19]: Posterity? JK. I'll remove them.

Comment (swainn) [reply .github/workflows/e2e_tests.yml]: Done. I'd like to keep the PostGIS versions for now though. I plan to get it working for 3.2 soon.

**Review by gagelarsen — APPROVED** [review]
Looks good

**Review by sdc50 — APPROVED** [review]

**Comment by sdc50** [comment]
@swainn It looks like the `requirements.txt` was removed. I'm interested in getting this on PyPI. What would that take?

**Comment by swainn** [comment]
> @swainn It looks like the `requirements.txt` was removed. I'm interested in getting this on PyPI. What would that take?

@sdc50 I did remove the file, but moved the requirements to setup.cfg. I'm planning to publish to PyPI and then submit a conda-forge recipe for it.

**Pull request merged**